ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/castor/algorithm/AutomatonSimulator.java

```diff
@@ -109,14 +114,32 @@ AutomatonPerimeterResultImpl simulateAutomatonState(State automatonState, Set<St
             return createFailedAutomatonPerimeterResult(automatonState, topoSimulationResult.getPerimeterResult(), topoSimulationResult.getActivatedNetworkActions(), "during");
         }
 
-        // II) Simulate range actions
-        RangeAutomatonSimulationResult rangeAutomatonSimulationResult = simulateRangeAutomatons(automatonState, curativeStates, network, preAutoPerimeterSensitivityAnalysis, topoSimulationResult.getPerimeterResult());
+        // II) Run auto search tree on available topological automatons
+        OptimizationPerimeter autoOptimizationPerimeter = AutoOptimizationPerimeter.build(automatonState, crac, network, raoParameters, topoSimulationResult.getPerimeterResult());
+        OptimizationResult autoSearchTreeResult = null;
+
+        PrePerimeterResult postAutoSearchTreeResult = topoSimulationResult.getPerimeterResult();
+
+        if (!autoOptimizationPerimeter.getNetworkActions().isEmpty()) {
+            autoSearchTreeResult = runAutoSearchTree(automatonState, network, stateTree, automatonTreeParameters, topoSimulationResult.getPerimeterResult(), autoOptimizationPerimeter, topoSimulationResult.getActivatedNetworkActions());
+            if (autoSearchTreeResult.getSensitivityStatus(automatonState) == ComputationStatus.FAILURE) {
+                return createFailedAutomatonPerimeterResult(automatonState, topoSimulationResult.getPerimeterResult(), autoSearchTreeResult.getActivatedNetworkActions(), "during");
+            }
+            applyRemedialActions(network, autoSearchTreeResult, automatonState);
+
+            PrePerimeterSensitivityAnalysis postAutoSearchTreeSensitivityAnalysis = new PrePerimeterSensitivityAnalysis(autoOptimizationPerimeter.getFlowCnecs(), Set.of(), raoParameters, toolProvider);
```

Comment:
Can you not use preAutoPerimeterSensitivityAnalysis? (defined at the beginning of the method)
Maybe it would need renaming if that's the case (it's already being used after applying forced topos so it should prob be renamed anyways)

Reply:
I'm not sure I fully understand, should I delete preAutoPerimeterSensitivityAnalysis and use another parameter instead?